internal/suggestions/suggest.go

```diff
@@ -0,0 +1,231 @@
+package suggestions
+
+import (
+	"context"
+	goerr "errors"
+	"fmt"
+	"github.com/manifoldco/promptui"
+	"github.com/speakeasy-api/openapi-generation/v2/pkg/errors"
+	"github.com/speakeasy-api/speakeasy/internal/auth"
+	"github.com/speakeasy-api/speakeasy/internal/log"
+	"github.com/speakeasy-api/speakeasy/internal/validation"
+	"go.uber.org/zap"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var ErrNoSuggestionFound = goerr.New("no suggestion found")
+
+func StartSuggest(ctx context.Context, schemaPath string, suggestionsConfig *Config, outputHints bool) error {
+	fmt.Println("Validating OpenAPI spec...")
+	fmt.Println()
+
+	schema, err := os.ReadFile(schemaPath)
+	if err != nil {
+		return fmt.Errorf("failed to read schema file %s: %w", schemaPath, err)
+	}
+
+	schema, err = ReformatFile(schema, DetectFileType(schemaPath))
+	if err != nil {
+		return fmt.Errorf("failed to reformat schema file %s: %w", schemaPath, err)
+	}
+
+	vErrs, vWarns, vInfo, err := validation.Validate(schema, schemaPath, outputHints)
+	if err != nil {
+		return err
+	}
+
+	printValidationSummary(vErrs, vWarns, vInfo)
+
+	toSuggestFor := vErrs
+	switch suggestionsConfig.Level {
+	case "warn":
+		toSuggestFor = append(toSuggestFor, vWarns...)
+		break
+	case "info":
+		toSuggestFor = append(append(toSuggestFor, vWarns...), vInfo...)
+		break
+	}
+
+	if len(toSuggestFor) > 0 {
+		err = Suggest(schema, schemaPath, toSuggestFor, *suggestionsConfig)
+		if err != nil {
+			fmt.Println(promptui.Styler(promptui.FGRed, promptui.FGBold)(fmt.Sprintf("cannot fetch llm suggestions: %s", err.Error())))
+			return err
+		}
+
+		if suggestionsConfig.OutputFile != "" && suggestionsConfig.AutoContinue {
```

Comment:
Why does `AutoContinue` have to be enabled for this log statement to be output?

Reply:
Because if it isn't then you'll have the y/n every step of the way

Reply:
I'm a bit confused, because I'm not sure how removing this changes the `y/n` step logic. Isn't that done here? https://github.com/speakeasy-api/speakeasy/blob/a0e8fbf0eb7cd8277b327ec7889da4254b8f5fff/internal/suggestions/suggestions.go#L85-L87